client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -511,116 +516,71 @@ def inject_ayon_environment(deadlinePlugin):
-        )
-        export_url = os.path.join(tempfile.gettempdir(), temp_file_name)
-        print(">>> Temporary path: {}".format(export_url))
-
-        add_kwargs = {
-            "envgroup": "farm",
-        }
-        # Support backwards compatible keys
-        for key, env_keys in (
-            ("project", ["AYON_PROJECT_NAME", "AVALON_PROJECT"]),
-            ("folder", ["AYON_FOLDER_PATH", "AVALON_ASSET"]),
-            ("task", ["AYON_TASK_NAME", "AVALON_TASK"]),
-            ("app", ["AYON_APP_NAME", "AVALON_APP_NAME"]),
-        ):
-            value = ""
-            for env_key in env_keys:
-                value = job.GetJobEnvironmentKeyValue(env_key)
-                if value:
-                    break
-            add_kwargs[key] = value
-
-        if not all(add_kwargs.values()):
-            raise RuntimeError((
-                "Missing required env vars: AYON_PROJECT_NAME,"
-                " AYON_FOLDER_PATH, AYON_TASK_NAME, AYON_APP_NAME"
-            ))
-
-        # Use applications addon arguments
-        # TODO validate if applications addon should be used
-        args = [
-            "--headless",
-            "addon",
-            "applications",
-            "extractenvironments",
-            export_url
-        ]
-
-        # staging requires passing argument
-        # TODO could be removed when PR in ayon-core starts to fill
-        #  'AYON_USE_STAGING' (https://github.com/ynput/ayon-core/pull/1130)
-        #  - add requirement for "core>=1.1.1" to 'package.py' when removed
-        settings_variant = job.GetJobEnvironmentKeyValue(
-            "AYON_DEFAULT_SETTINGS_VARIANT"
-        )
-        if settings_variant == "staging":
-            args.append("--use-staging")
-
-        # Backwards compatibility for older versions
-        legacy_args = [
-            "--headless",
-            "extractenvironments",
-            export_url
-        ]
-
-        for key, value in add_kwargs.items():
-            args.extend(["--{}".format(key), value])
-            # Legacy arguments expect '--asset' instead of '--folder'
-            if key == "folder":
-                key = "asset"
-            legacy_args.extend(["--{}".format(key), value])
-
-        environment = {
-            "AYON_SERVER_URL": ayon_server_url,
-            "AYON_API_KEY": ayon_api_key,
-            "AYON_BUNDLE_NAME": ayon_bundle_name,
-        }
-
-        for key in ("AYON_USE_STAGING", "AYON_IN_TESTS"):
-            value = job.GetJobEnvironmentKeyValue(key)
-            if value:
-                environment[key] = value
+            )
 
-        for env, val in environment.items():
-            # Add the env var for the Render Plugin that is about to render
-            deadlinePlugin.SetEnvironmentVariable(env, val)
-            # Add the env var for current calls to `DeadlinePlugin.RunProcess`
-            deadlinePlugin.SetProcessEnvironmentVariable(env, val)
+        output_dir = _get_output_dir(job)
```

Comment:
Can we move this into the `if shared_env_group:` please? Also, this will error jobs if they do not have any of `["OutputFilePath", "Output", "SceneFile"]` which may very well be. I think the choice for `SceneFile` isn't bullet proof either- because it's `SceneFile` for Maya, Houdini, Nuke, After Effects, Cinema4D, 3dsmax but `FlowFile` for Fusion 🤷